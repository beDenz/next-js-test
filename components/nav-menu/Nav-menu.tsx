import styles from './Nav-menu.module.scss'
import Link from 'next/link'

interface MenuItem {
  id: number
  link: string
  title: string
}

const NavMenuItem: React.FC<MenuItem> = ({ id, link, title }) => {
  return (
    <li className={styles['nav-menu-item']}>
      <Link href={link}>{title}</Link>
    </li>
  )
}

const NavMenu: React.FC<{ list: MenuItem[] }> = ({ list }): JSX.Element => {
  return (
    <ul className={styles.navMenu}>
      {list.map(({ id, link, title }) => (
        <NavMenuItem id={id} link={link} title={title} key={id} />
      ))}
    </ul>
  )
}

export { NavMenu }
