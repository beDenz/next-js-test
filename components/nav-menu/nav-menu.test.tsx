import React from 'react'
import { render } from 'test/testUtils'
import { create } from 'react-test-renderer'
// import { render, fireEvent } from '../testUtils'
import { NavMenu } from './Nav-menu'

const menu = [
  {
    id: 1,
    link: '/',
    title: 'home',
  },
  {
    id: 2,
    link: '/pages',
    title: 'pages',
  },
  {
    id: 3,
    link: '/about',
    title: 'about',
  },
  {
    id: 4,
    link: '/ssr',
    title: 'ssr',
  },
  {
    id: 5,
    link: '/',
    title: 'login',
  },
]

describe('Navigation Menu', () => {
  it('naavigation menu matches snapshot', () => {
    const { asFragment } = render(<NavMenu list={menu} />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('navigation menu current item', () => {
    const rerender = create(<NavMenu list={menu} />)

    expect(rerender.root.findAllByType('li').length).toBe(menu.length)
  })
})
