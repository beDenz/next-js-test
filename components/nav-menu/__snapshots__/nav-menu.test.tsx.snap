// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Menu naavigation menu matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="navMenu"
  >
    <li
      class="nav-menu-item"
    >
      <a
        href="/"
      >
        home
      </a>
    </li>
    <li
      class="nav-menu-item"
    >
      <a
        href="/pages"
      >
        pages
      </a>
    </li>
    <li
      class="nav-menu-item"
    >
      <a
        href="/about"
      >
        about
      </a>
    </li>
    <li
      class="nav-menu-item"
    >
      <a
        href="/ssr"
      >
        ssr
      </a>
    </li>
    <li
      class="nav-menu-item"
    >
      <a
        href="/"
      >
        login
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
